src/Database/LSMTree/Internal/MergingRun.hs

```diff
@@ -258,6 +258,10 @@ unsafeNew mergeDebt (SpentCredits spentCredits)
 
 -- | Create references to the runs that should be queried for lookups.
 -- In particular, if the merge is not complete, these are the input runs.
+--
+-- TODO: This interface doesn't work well with the action registry. Just doing
+-- @withRollback reg (duplicateRuns mr) (mapM_ releaseRef)@ isn't exception-safe
+-- since if one of the @releaseRef@ calls fails, the following ones aren't run.
```

Comment:
I think it works fine with the action registry, but we need a `releaseRuns` function that (once and for all) releases a vector of runs in an exception safe manner, instead of using `V.mapM_ releaseRef`